python/test/test_polars.py

```diff
@@ -444,11 +445,17 @@ def test_polars_non_wrapping():
     # only calls that return a LazyFrame or LazyGroupBy are wrapped
     assert context.query().explain() == 'DF ["A"]; PROJECT */1 COLUMNS; SELECTION: None'
     assert context.query().collect_schema() == {"A": pl.String}
-    serial = context.query().with_columns(pl.col("A") + 2).serialize(format="json")
-    assert serial.startswith('{"HStack":')
+
     # for coverage: attribute access works properly
     context.query()._ldf
 
+    # serialize/deserialize roundtrip
+    serial_query = context.query().select(dp.len())
+    serial_query._lf_plan = pl.LazyFrame.deserialize(io.BytesIO(serial_query.serialize())) # type: ignore
+    rel = serial_query.release().collect()
+    assert rel.schema == pl.Schema({"len": pl.UInt32})
+    assert isinstance(rel["len"][0], int)
```

Comment:
These assertions aren't going the way I'd expect. Is there a reason not to take a query, serialize it, deserialize it, and then compare it to the original... or release both, and make sure they behave the same way?